virttest/test_setup.py

```diff
@@ -2029,16 +2030,50 @@ def stop(self):
         self._compress_log()
 
 
-def disable_smt():
+def disable_smt(params=None):
     """
     Checks whether smt is on, if so disables it in PowerPC system
-    This function is used in env_process to check & disable smt in Power8
+    This function is used in env_process and in libvirt.py to check
+    & disable smt in Power8 for local and remote machine respectively.
+
+    :param params: Test params dict for remote machine login details
     """
-    smt_output = process.system_output("ppc64_cpu --smt", shell=True).strip()
-    if smt_output != "SMT is off":
-        try:
-            utils_misc.verify_running_as_root()
-            process.run("ppc64_cpu --smt=off", verbose=True, shell=True)
-            logging.debug("SMT turned off successfully")
-        except process.CmdError, info:
-            raise exceptions.TestSetupFail("VM can not be started :%s", info)
+    smt_output = ""
+    cmd = "ppc64_cpu --smt"
+    if params:
+        server_ip = params.get("server_ip")
```

Comment:
please use param["server_ip"] to avoid server_ip missed in test parms, 'server_pwd' also need to update.